test/cpp/interop/backend_metrics_lb_policy.cc

```diff
@@ -0,0 +1,240 @@
+                     RefCountedPtr<SubchannelPicker> picker) override {
+      parent_->channel_control_helper()->UpdateState(
+          state, status,
+          MakeRefCounted<Picker>(std::move(picker),
+                                 parent_->load_report_tracker_));
+    }
+
+    void RequestReresolution() override {
+      parent_->channel_control_helper()->RequestReresolution();
+    }
+
+    absl::string_view GetAuthority() override {
+      return parent_->channel_control_helper()->GetAuthority();
+    }
+
+    grpc_event_engine::experimental::EventEngine* GetEventEngine() override {
+      return parent_->channel_control_helper()->GetEventEngine();
+    }
+
+    void AddTraceEvent(TraceSeverity severity,
+                       absl::string_view message) override {
+      parent_->channel_control_helper()->AddTraceEvent(severity, message);
+    }
+
+   private:
+    RefCountedPtr<BackendMetricsLbPolicy> parent_;
+  };
+
+  class SubchannelCallTracker : public SubchannelCallTrackerInterface {
+   public:
+    explicit SubchannelCallTracker(LoadReportTracker* load_report_tracker)
+        : load_report_tracker_(load_report_tracker) {}
+
+    void Start() override {}
+
+    void Finish(FinishArgs args) override {
+      load_report_tracker_->RecordPerRpcLoadReport(
+          args.backend_metric_accessor->GetBackendMetricData());
+    }
+
+   private:
+    LoadReportTracker* load_report_tracker_;
+  };
+
+  void ShutdownLocked() override { delegate_.reset(); }
+
+  OrphanablePtr<LoadBalancingPolicy> delegate_;
+  LoadReportTracker* load_report_tracker_;
+};
+
+class BackendMetricsLbPolicyFactory
+    : public grpc_core::LoadBalancingPolicyFactory {
+ private:
+  class BackendMetricsLbPolicyFactoryConfig
+      : public LoadBalancingPolicy::Config {
+   private:
+    absl::string_view name() const override {
+      return kBackendMetricsLbPolicyName;
+    }
+  };
+
+  absl::string_view name() const override {
+    return kBackendMetricsLbPolicyName;
+  }
+
+  OrphanablePtr<LoadBalancingPolicy> CreateLoadBalancingPolicy(
+      LoadBalancingPolicy::Args args) const override {
+    return grpc_core::MakeOrphanable<BackendMetricsLbPolicy>(std::move(args));
+  }
+
+  absl::StatusOr<RefCountedPtr<LoadBalancingPolicy::Config>>
+  ParseLoadBalancingConfig(const grpc_core::Json& /*json*/) const override {
+    return MakeRefCounted<BackendMetricsLbPolicyFactoryConfig>();
+  }
+};
+}  // namespace
+
+void RegisterBackendMetricsLbPolicy(CoreConfiguration::Builder* builder) {
+  builder->lb_policy_registry()->RegisterLoadBalancingPolicyFactory(
+      std::make_unique<BackendMetricsLbPolicyFactory>());
+}
+
+void LoadReportTracker::SetupOnChannel(ChannelArguments* arguments) {
+  arguments->SetPointer(kMetricsTrackerAttribute, this);
+}
+
+void LoadReportTracker::RecordPerRpcLoadReport(
+    const grpc_core::BackendMetricData* backend_metric_data) {
+  if (backend_metric_data != nullptr) {
```

Comment:
I don't think we need to check whether it's null here, because `BackendMetricDataToOrcaLoadReport()` will return nullopt in that case.  I think we can just unconditionally add the result of `BackendMetricDataToOrcaLoadReport()` to the queue.

Reply:
Done